src/persist/src/operators/stream.rs

```diff
@@ -1103,150 +1099,59 @@ mod tests {
         let actual_seals: Vec<_> = listen_rx.try_iter().collect();
 
         // Assert that:
-        //  a) We don't seal primary when condition has not sufficiently advanced.
-        //  b) Condition is sealed before primary for the same timestamp OR
-        //     primary and condition got swapped when reading from two channels and a
-        //     primary seal at t is just before the condition seal at a time t' >= t.
-        //  c) We seal up, even when never receiving any data.
-        //  d) Seals happen in timestamp order.
+        //  a) We seal up, even when never receiving any data.
```

Comment:
I think the only other invariant that `atomic_seal` holds is "either all of the relevant collections get sealed to a time t or none of them do, but you never have a subset of collections getting sealed"?

Reply:
I'll extend the test to cover that. :ok_hand: